publisher.rb

```diff
@@ -1,33 +1,55 @@
 require 'aws-sdk-sqs'
 require_relative 'load_aws'
 require_relative 'aws_sqs_client'
+require_relative 'q_it_errors'
 
 class Publisher
   
   attr_reader :sqs, :sleep_period, :queue_name
 
   def initialize(options)
     @sqs = AwsSQSClient.new.client
-    @sleep_period = valid_sleep_period?(options[1]) ? options[1].to_i : 5
     @queue_name = options[0]
+    @sleep_period = options[1]
+  end
+
+  def self.validate(options)
+    raise QItArgumentError, "Incorrect number of arguments. Expected 2 got #{options.count}" if options.count != 2
+    
+    queue_name = options[0]
+    sleep_period = options[1]
+    valid_queue_name?(queue_name) && valid_sleep_period?(sleep_period)
```

Comment:
No need to set local variable names `queue_name` and `sleep_period` just use the `options` array.

Reply:
We need to. Otherwise it will return `nil`, and we are not doing a `nil` check in L19. So it will still execute with wrong values.